src/Interfaces/LoginManager.vala

```diff
@@ -68,4 +70,22 @@ public class About.LoginManager : Object {
 
         return true;
     }
+
+    public void set_reboot_to_firmware_setup () {
+        try {
+            interface.set_reboot_to_firmware_setup (true);
+        } catch (Error e) {
+            warning ("Could not connect to login interface: %s", e.message);
```

Comment:
It seems like we don't want to fail silently here. If we can't set this flag then we should abort the system restart and throw an error to the user that we're unable to restart into the firmware setup screen. Otherwise we're restarting them for no reason with no explanation